server.js

```diff
@@ -1,16 +1,117 @@
 const express = require("express");
 const app = express();
 const port = 3000;
+const path = require('path');
 const studentRoutes = require("./src/students/routes");
+const bodyParser = require('body-parser');
+console.log("Created server");
+const html = require('html');
+const pug = require('pug');
 
-app.get("/" , (req,res) => {
+app.set('view engine', 'pug');
+app.get('/', (req, res) => {
+    // Render the Pug template
+    res.render('index', {
+      title: 'My Website',
+    });
+  });
+app.use(bodyParser.json());
+/*app.get("/" , (req,res) => {
     res.send("Hello World");
 });
+console.log("Created server");
+//app.use(express.json());
+app.use(express.static(path.join(__dirname, 'public')));
 
-app.use(express.json());
-
-app.use("/students", studentRoutes)
-
+app.get('/', async(req, res) => {
+    res.render('index.html')
+});
+//app.use("/students", studentRoutes)
+app.post('/submit', studentRoutes);
 app.listen(port, () => {
     console.log(`App is listening at port ${port}`)
+}); */
+const pg = require("pg")
+
+const pool = new pg.Pool({
+    user: "harwinderkaur",
+    password: "",
+    database: "students",
+    host: "localhost",
+    port: 5432 // Default PostgreSQL port
+});
+// Create a route for the registration form
+app.get('/register', (req, res) => {
+  // Render the registration form
+  res.render('registration.html');
+});
+
+// Create a route for the signup process
+app.post('/signup', async (req, res) => {
+  // Get the user input
+  const username = req.body.username;
+  const password = req.body.password;
+  const name = req.body.name;
+  const dob = req.body.dob;
+
+  // Validate the user input
+  // ...
+
+  // Hash the password
+  const hashedPassword = await bcrypt.hash(password, 10);
```

Comment:
Why do we need to hash password?

What if you database get hacked. How can hacker recover the password? 

Why do we need complex rules for password and how does that make it secure?

How would you add rules to password?